### ISSUE TITLE
pom.xml:348-352: Enable RuleAssertionMessage. This rule...

### ISSUE THREAD

**Issue by 0pdd** [objectionary/eo#2297]
The puzzle `2171-74d15acf` from #2171 has to be resolved: 

https://github.com/objectionary/eo/blob/992706f77927a94ed361e3fd2e4771e3cea6bd07/pom.xml#L348-L352

The puzzle was created by rultor on 21-Jul-23. 

Estimate: 30 minutes,  role: DEV. 

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

**Comment by yegor256** [comment]
@volodya-lombrozo check this out: https://docs.pmd-code.org/latest/pmd_rules_apex_bestpractices.html#apexunittestclassshouldhaveasserts I suggest removing this check from JTCop

**Comment by volodya-lombrozo** [comment]
@yegor256 I'm not sure about "removing". Maybe it's better just to disable that check by default. What do you think?
As for `PMD` check - Why does not this check work in `eo` project? Does `qulice` uses the old version of `PMD`?

**Comment by c71n93** [comment]
@volodya-lombrozo @yegor256 for now `qulice` has check for assertions without messages and now all assertions contain messages (most of them are stubs). So I tried to enable the `JTCOP.RuleAssertionMessage` and faced with some problems. For example:
```
Method 'checksTypoPacks' has assertion without message: 'MatcherAssert.assertThat(EoSyntaxTest.EMPTY_MSG, XhtmlMatchers.xhtml(xml.toString()), XhtmlMatchers.hasXPaths("/program/errors/error/@line"))'. 
Please add the explanation message to make the test more readable.
```
Here, assertion contains a stub message (`EoSyntaxTest.EMPTY_MSG`), however this assertion still fails for some reason.

I think this todo should be removed, because main problem was solved, but first we need to resolve such contradiction. What do you think about disabling this jtcop rule by default?

**Comment by volodya-lombrozo** [comment]
@c71n93 Can you just create an issue in the [jtcop](https://github.com/volodya-lombrozo/jtcop) repository? I will fix it.

**Comment by c71n93** [comment]
@volodya-lombrozo It seems like theres another bug. For example:
```
[ERROR] The test class OptimizeMojoTest.java (/Users/c71n93/Programming/EOLANG/eo/eo-maven-plugin/src/test/java/org/eolang/maven/OptimizeMojoTest.java:) has encountered some problems. Please review the results for more information.
[ERROR]         2) Method choosesTransformerFactoryInConcurrentEnvironment doesn't have assertion statements.
[ERROR]         Please add at least one assertion statement to the test method.
[ERROR]         You can also ignore the rule by adding @SuppressWarnings("JTCOP.RuleAssertionMessage") annotation.
[ERROR]         Rule: RuleAssertionMessage.
```
The test looks like this:
```java
    @Test
    void choosesTransformerFactoryInConcurrentEnvironment() {
        for (final Class<? extends TransformerFactory> clazz : IntStream.range(0, 100).parallel()
            .mapToObj(i -> TransformerFactory.newInstance().getClass())
            .collect(Collectors.toList())) {
            MatcherAssert.assertThat(
                "TO ADD ASSERTION MESSAGE",
                clazz,
                Matchers.typeCompatibleWith(TransformerFactoryImpl.class)
            );
        }
    }
```
Will you fix it? It looks like check ignores asserts inside of loops.

**Comment by c71n93** [comment]
@volodya-lombrozo @yegor256 also there was cases, that was not detected by `qulice`, but detected by `jtcop`. For example:
```java
    @Test
    void failsOnTimeout(@TempDir final Path temp) {
        Assertions.assertThrows(
            IllegalStateException.class,
            () -> new FakeMaven(temp)
                .withHelloWorld()
                .with("timeout", 0)
                .execute(InfiniteMojo.class),
        );
    }
```
For some reasons `qulice` ignores `Assertions.assertThrows`. I start resolving such errors.
@volodya-lombrozo do you mind to reassign this issue on me?

**Comment by volodya-lombrozo** [comment]
@c71n93 

> Will you fix it? It looks like check ignores asserts inside of loops.

For sure. Again, could report a bug in [jtcop](https://github.com/volodya-lombrozo/jtcop), please?

> @volodya-lombrozo do you mind to reassign this issue on me?

Yes, of course.

**Comment by c71n93** [comment]
@volodya-lombrozo could you check this one too, please https://github.com/volodya-lombrozo/jtcop/issues/357. It also related to this ticket.

**Comment by volodya-lombrozo** [comment]
@c71n93 Yes, I'll take a look.

**Comment by 0pdd** [comment]
@0pdd 3 puzzles [#3130](https://github.com/objectionary/eo/issues/3130), [#3131](https://github.com/objectionary/eo/issues/3131), [#3132](https://github.com/objectionary/eo/issues/3132) are still not solved.

**Comment by 0pdd** [comment]
@0pdd 3 puzzles [#3131](https://github.com/objectionary/eo/issues/3131), [#3132](https://github.com/objectionary/eo/issues/3132), [#3689](https://github.com/objectionary/eo/issues/3689) are still not solved; solved: [#3130](https://github.com/objectionary/eo/issues/3130).

**Comment by 0pdd** [comment]
@0pdd 2 puzzles [#3131](https://github.com/objectionary/eo/issues/3131), [#3132](https://github.com/objectionary/eo/issues/3132) are still not solved; solved: [#3130](https://github.com/objectionary/eo/issues/3130), [#3689](https://github.com/objectionary/eo/issues/3689).

**Comment by 0pdd** [comment]
@0pdd 2 puzzles [#3131](https://github.com/objectionary/eo/issues/3131), [#4240](https://github.com/objectionary/eo/issues/4240) are still not solved; solved: [#3130](https://github.com/objectionary/eo/issues/3130), [#3132](https://github.com/objectionary/eo/issues/3132), [#3689](https://github.com/objectionary/eo/issues/3689).
